lib/GHCup/Errors.hs

```diff
@@ -34,7 +35,136 @@ import           URI.ByteString
 
 import qualified Data.Map.Strict               as M
 import qualified Data.Text                     as T
-
+import           Data.Data (Proxy(..))
+
+
+
+allHFError :: String
```

Comment:
This seems difficult to maintain over time. But I guess it's no worse than any other documentation.

Reply:
I think they used an injective type family to ensure uniqueness of codes in GHC - I wonder whether something like that could happen here, given that you're manually specifying them? Probably not worth it.